integration/generic-metadata/hero.ts

```diff
@@ -303,21 +303,21 @@ export class HeroServiceClientImpl implements HeroService {
     this.FindOneVillain = this.FindOneVillain.bind(this);
     this.FindManyVillain = this.FindManyVillain.bind(this);
   }
-  FindOneHero(request: HeroById): Promise<Hero> {
+  FindOneHero(request: HeroById, metadata?: Metadata): Promise<Hero> {
```

Comment:
Just checking my understanding, but given this is the `...ClientImpl`, is the idea of exposing `metadata` here that it allows clients to pass in something like `{ userId: 123, authToken: 345 }` into the client calls?

Fwiw/if so, I think historically I'd assumed that sort of metadata would be added by the `Rpc` implementation, instead of having to be passed on every single client method invocation.

But curious to hear if that makes sense/how you're using the `metadata` param on client calls.

Reply:
We generally pass a metadata factory to `Rpc` implementation but we have a use case, where the backend implementation expects additional metadata, so the idea here is to be able to override them.